internal/provider/resources/user_api_key_test.go

```diff
@@ -0,0 +1,64 @@
+package resources_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/statecheck"
+	"github.com/prefecthq/terraform-provider-prefect/internal/testutils"
+)
+
+func fixtureAccUserAPIKeyCreate(userID, name string) string {
+	return fmt.Sprintf(`
+resource "prefect_user_api_key" "test" {
+  user_id = "%s"
+  name = "%s"
+}
+`, userID, name)
+}
+
+func fixtureAccUserAPIKeyRecreate(userID, name string, expiration time.Time) string {
+	return fmt.Sprintf(`
+resource "prefect_user_api_key" "test" {
+  user_id = "%s"
+  name = "%s"
+	expiration = "%s"
+}
+`, userID, name, expiration.Format(time.RFC3339))
+}
+
+//nolint:paralleltest // we use the resource.ParallelTest helper instead
+func TestAccResource_user_api_key(t *testing.T) {
+	resourceName := "prefect_user_api_key.test"
+	userID := os.Getenv("ACC_TEST_USER_RESOURCE_ID")
+	name := testutils.TestAccPrefix + acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
```

Comment:
```suggestion
	name := testutils.NewRandomPrefixedString()
```

We have a helper for this one 👍🏼

Reply:
of course we do 😅 updating now

Reply:
updated here https://github.com/PrefectHQ/terraform-provider-prefect/pull/418/commits/2bb7e921ea78cb9bde015796a36e9301d8ed8fdc